tests/test_access_module.py

```diff
@@ -0,0 +1,132 @@
+    oracleProvider,
+    riskpoolKeeper,
+):
+    instanceOperatorService = instance.getInstanceOperatorService()
+    instanceService = instance.getInstanceService()
+    ioDict = {'from': owner}
+    
+    (poRole, opRole, rkRole) = getRoles(instanceService)
+
+    instanceOperatorService.grantRole(poRole, productOwner, ioDict)
+    instanceOperatorService.grantRole(opRole, oracleProvider, ioDict)
+    instanceOperatorService.grantRole(rkRole, riskpoolKeeper, ioDict)
+
+    assert instanceService.hasRole(poRole, productOwner)
+    assert instanceService.hasRole(opRole, oracleProvider)
+    assert instanceService.hasRole(rkRole, riskpoolKeeper)
+
+    instanceOperatorService.revokeRole(poRole, productOwner, ioDict)
+    instanceOperatorService.revokeRole(opRole, oracleProvider, ioDict)
+    instanceOperatorService.revokeRole(rkRole, riskpoolKeeper, ioDict)
+
+    assert not instanceService.hasRole(poRole, productOwner)
+    assert not instanceService.hasRole(opRole, oracleProvider)
+    assert not instanceService.hasRole(rkRole, riskpoolKeeper)
+
+
+def test_role_creation(
+    instance,
+    owner,
+    productOwner,
+    oracleProvider,
+    riskpoolKeeper,
+):
+    instanceOperatorService = instance.getInstanceOperatorService()
+    instanceService = instance.getInstanceService()
+    ioDict = {'from': owner}
+
+    NEW_ROLE = keccak256('NEW_ROLE')
+
+    # check that unknown roles cannot be granted
+    with brownie.reverts('ERROR:ACL-002:ROLE_UNKNOWN_OR_INVALID'):
+        instanceOperatorService.grantRole(NEW_ROLE, productOwner, ioDict)
+
+    # check that a non instance operator cannot create new role
+    with brownie.reverts('ERROR:IOS-001:NOT_INSTANCE_OPERATOR'):
+        instanceOperatorService.createRole(NEW_ROLE, {'from': productOwner})
+
+    # role creation
+    instanceOperatorService.createRole(NEW_ROLE, ioDict)
+
+    assert not instanceService.hasRole(NEW_ROLE, productOwner)
+
+    # grant newly created role
+    instanceOperatorService.grantRole(NEW_ROLE, productOwner, ioDict)
+
+    # check granting
+    assert instanceService.hasRole(NEW_ROLE, productOwner)
+
+
+def test_role_invalidation(
+    instance,
+    owner,
+    productOwner,
+    oracleProvider,
+    riskpoolKeeper,
+):
+    instanceOperatorService = instance.getInstanceOperatorService()
+    instanceService = instance.getInstanceService()
+    ioDict = {'from': owner}
+    
+    (poRole, opRole, rkRole) = getRoles(instanceService)
+
+    instanceOperatorService.grantRole(poRole, productOwner, ioDict)
+
+    # check that a non instance operator cannot invalidate a role
+    with brownie.reverts('ERROR:IOS-001:NOT_INSTANCE_OPERATOR'):
+        instanceOperatorService.invalidateRole(poRole, {'from': productOwner})
+
+    NEW_ROLE = keccak256('NEW_ROLE')
+
+    with brownie.reverts('ERROR:ACL-004:ROLE_UNKNOWN_OR_INVALID'):
+        instanceOperatorService.invalidateRole(NEW_ROLE, ioDict)
+
+    instanceOperatorService.invalidateRole(poRole, ioDict)

```

Comment:
We should also make sure this can be reverted again ... in case someone uses this accidentally it can hurt.